frontend/src/components/widget/MasterWidget.js

```diff
@@ -102,31 +97,15 @@ class MasterWidget extends Component {
       isAdvanced = false,
       viewId,
     } = this.props;
-    const numberField = isNumberField(widgetType);
 
-    if (widgetType === 'Quantity' && value === '') {
-      value = null;
-    } else if (numberField && !value) {
-      value = '0';
-    }
-
-    let { entity } = this.props;
-    let currRowId = rowId;
+    let entity = viewId ? 'documentView' : this.props.entity;
+    value = formatValueByWidgetType({ widgetType, value });
+    let currRowId = rowId === 'NEW' ? relativeDocId : rowId;
     let ret = null;
-    let isEdit = false;
-
-    if (rowId === 'NEW') {
-      currRowId = relativeDocId;
-    }
+    let isEdit = viewId ? true : false;
 
-    if (widgetType !== 'Button') {
+    widgetType !== 'Button' &&
```

Comment:
Does this really give us anything ?

Reply:
Yes. One line of code less. Using such through the app whenever is possible it will reduce the overall lines of code written.